internal/rbd/rbd_util.go

```diff
@@ -1946,3 +1964,22 @@ func (rv *rbdVolume) setSnapshotMetadata(parameters map[string]string) error {
 
 	return nil
 }
+
+// unsetSnapshotMetadata unset snapshot-name/snapshot-namespace/snapshotcontent-name metadata
```

Comment:
you probably do not need this function if you pass the keys as a `[]string`. It is the same as `unsetVolumeMetadata()`?

Reply:
the function is almost completely duplicated, if there really is a need for separate functions (hint: there probably is not), use a helper to improve maintainability.

Reply:
In addition to my previous comment,
Calling unsetVolumeMetadata() for Snapshot metadata looks a bit awkward to me, CodeReadabilty++

Reply:
just rename the function to `SetMetadataList(data map[string]string)` and `UnsetMetadataList(keys []string)`? The object for both Snapshot and Volume are `rbdVolume`, so no need to have Volume or Snapshot in the function names. CodeReadabilty++ and CodeMaintainability++

Reply:
@nixpanic Done now, PTAL